doc/source/text.rst

```diff
@@ -205,6 +217,52 @@ and optional groups like
 
 can also be used.
 
+When ``str.extract`` method is used on an ``Index``, it returns a
+``DataFrame`` with as many columns as capture groups.
```

Comment:
it is in the examples of the doc-string but I will also add a specific description under Returns: "Any capture group names in regular expression pat will be used for column names; otherwise capture group numbers will be used." I also added that to text.rst.